ios/KaMPKitiOS/ViewController.swift

```diff
@@ -79,3 +88,31 @@ extension BreedsViewController: BreedCellDelegate {
         adapter.updateBreedFavorite(breed: breed)
     }
 }
+
+enum DataStateEnum<T> {
+    case Success(_ data: T)
+    case Error(_ error: Error)
+    case Loading
+    case Empty
+}
+
+func DataStateToEnum<T>(_ dataState: DataState<T>) -> DataStateEnum<T> {
```

Comment:
This seems fine to me? It has a little code smell but I don't have a better approach. You can also do an `if let` to prevent any crashes but that's my only comment.